lightning/src/ln/peer_handler.rs

```diff
@@ -1238,7 +1256,14 @@ impl<Descriptor: SocketDescriptor, CM: Deref, RM: Deref, OM: Deref, L: Deref, CM
 					},
 					InitSyncTracker::ChannelsSyncing(c) if c == 0xffff_ffff_ffff_ffff => {
 						if let Some(msg) = self.message_handler.route_handler.get_next_node_announcement(None) {
-							self.enqueue_message(peer, &msg);
+							let (min, max) = peer.sent_gossip_timestamp_filter.as_ref().map_or((0, u32::MAX), |f| {
+								(f.first_timestamp, f.first_timestamp.saturating_add(f.timestamp_range))
+							});
+
+							if msg.contents.timestamp >= min && msg.contents.timestamp <= max {
+								self.enqueue_message(peer, &msg);
```

Comment:
> Based on that, it flips between either a full sync or new gossip relay only.

Reading the following comment to try to understand how `gossip_timestamp_filter` works.

https://github.com/lightningdevkit/rust-lightning/blob/0c31021c6f96c20114420b62e9512d22af311461/lightning/src/routing/gossip.rs#L650-L663

IIUC, it seems with this PR we either send the full dump or not based on the timestamp. And if the newly added check doesn't pass, we won't send *any* gossip. It doesn't seem to be the case that we'd even relay new gossip. This is because `should_forward_channel_announcement` and `should_forward_node_announcement` will return `false` when `sent_gossip_timestamp_filter` is `false`.